File: src/java/_1759_Count_Number_of_Homogenous_Substrings.java
class Solution {
    public int countHomogenous(String s) {
        int MOD = (int) Math.pow(10, 9) + 7;
        long ans = 0;
        long length = 0;
        char lastSeen = s.charAt(0);

        for (char c : s.toCharArray()) {
            if (c == lastSeen) {
                length += 1;
            } else {
                ans = (ans + length * (length + 1) / 2) % MOD;
                lastSeen = c;
                length = 1;
            }
        }

        ans = (ans + length * (length + 1) / 2) % MOD;

        return (int) ans;
    }
}

File: src/python/_1759_Count_Number_of_Homogenous_Substrings.py
class Solution:
    def countHomogenous(self, s: str) -> int:
        ans = 0
        length = 0
        lastSeen = s[0]

        for c in s:
            if c == lastSeen:
                length += 1
            else:
                ans += length * (length + 1) // 2
                lastSeen = c
                length = 1

        ans += length * (length + 1) // 2

        return ans % (10**9 + 7)
